File: sol/solution/0001-0100/0004. Median of Two Sorted Arrays/Solution.cs
public class Solution
{
    public double FindMedianSortedArrays(int[] nums1, int[] nums2)
    {
        if (nums1.Length > nums2.Length)
        {
            return FindMedianSortedArrays(nums2, nums1);
        }
        int x = nums1.Length, y = nums2.Length;
        int low = 0, high = x;
        while (low <= high)
        {
            int partitionX = (low + high) / 2;
            int partitionY = (x + y + 1) / 2 - partitionX;
            int maxX = (partitionX == 0) ? int.MinValue : nums1[partitionX - 1];
            int minX = (partitionX == x) ? int.MaxValue : nums1[partitionX];
            int maxY = (partitionY == 0) ? int.MinValue : nums2[partitionY - 1];
            int minY = (partitionY == y) ? int.MaxValue : nums2[partitionY];
            if (maxX <= minY && maxY <= minX)
            {
                if ((x + y) % 2 == 0)
                {
                    return (Math.Max(maxX, maxY) + Math.Min(minX, minY)) / 2.0;
                }
                else
                {
                    return Math.Max(maxX, maxY);
                }
            }
            else if (maxX > minY)
            {
                high = partitionX - 1;
            }
            else
            {
                low = partitionX + 1;
            }
        }
        return 0.0;
    }
}

File: sol/solution/0001-0100/0004. Median of Two Sorted Arrays/Solution.ts
function findMedianSortedArrays(nums1: number[], nums2: number[]): number {
    if (nums1.length > nums2.length) return findMedianSortedArrays(nums2, nums1);
    let x = nums1.length, y = nums2.length;
    let low = 0, high = x;
    while (low <= high) {
        const partitionX = Math.floor((low + high) / 2);
        const partitionY = Math.floor((x + y + 1) / 2) - partitionX;
        const maxX = partitionX === 0 ? -Infinity : nums1[partitionX - 1];
        const minX = partitionX === x ? Infinity : nums1[partitionX];
        const maxY = partitionY === 0 ? -Infinity : nums2[partitionY - 1];
        const minY = partitionY === y ? Infinity : nums2[partitionY];
        if (maxX <= minY && maxY <= minX) {
            if ((x + y) % 2 === 0)
                return (Math.max(maxX, maxY) + Math.min(minX, minY)) / 2;
            else
                return Math.max(maxX, maxY);
        } else if (maxX > minY)
            high = partitionX - 1;
        else
            low = partitionX + 1;
    }
    return 0;
}